ext/mbstring/mbstring.c

```diff
@@ -761,7 +761,7 @@ static PHP_INI_MH(OnUpdate_mbstring_http_input)
 		php_error_docref("ref.mbstring", E_DEPRECATED, "Use of mbstring.http_input is deprecated");
 	}
 
-	if (!new_value || !ZSTR_VAL(new_value)) {
+	if (!new_value) {
```

Comment:
Did you change this deliberately? If so, the implementation of the function could be simplified; basically to an `if-else`, then swap the branches, then drop the `else`, since the `then` branch guards.

Reply:
Yeah, it produced a compiler warning (as the check is redundant) but I didn't want to change anything else.

Reply:
Is that check really redundant? What about an empty `zend_string` (or more precisely, a `zend_string` where the first byte is NUL)?

Reply:
@cmb69 Then it's still decayed to a pointer that points to a `'\0'`.

Reply:
Just to demonstrate: https://godbolt.org/z/6x9z7d8a9

Reply:
Oh, you're right! However, I assume this is a bug stemming from the time we changed from `char*` to `zend_string*`, and it should actually be
````
if (!new_value || !ZSTR_VAL(new_value)[0]) {
````
Or to match the implementation of `PHP_INI_MH(OnUpdate_mbstring_http_output)`:
````
if (new_value == NULL || ZSTR_LEN(new_value) == 0) {
````

Reply:
Sounds like adding a new regression test is in order.

Reply:
> I assume this is a bug stemming from the time we changed from `char*` to `zend_string*`

Doesn't look like it. It was added here: https://github.com/php/php-src/commit/379d9a1cfc64626d3d7627aaf53c22cb950ec343 But maybe it was just always wrong.

Reply:
Just tried checking what the manual says about `mbstring.http_input`: https://www.php.net/manual/en/mbstring.configuration.php#ini.mbstring.http-input

It doesn't document what is "supposed" to happen if `mbstring.http_input` is set to an empty string. I suggest we stick to whatever the historical behavior was.

Reply:
Yeah, might be best for now. After all, these deprecated INI settings were supposed to be removed long ago.

Reply:
I'll merge this without that commit for now.